test/stdlib/NSKeyedArchival.swift

```diff
@@ -35,8 +35,13 @@ func expectNoError(_ verb: () throws -> ()) {
 func archive(_ archival: (_: NSKeyedArchiver) throws -> Void) rethrows ->  Data {
     let data = NSMutableData()
     let ka = NSKeyedArchiver(forWritingWith: data)
-    defer { ka.finishEncoding() }
-    try archival(ka)
+    do {
```

Comment:
One thing you can do to avoid the multiple `finishEncoding()` calls, I remembered, is put the `defer` into the `do` itself (since `defer`s are scoped):

```swift
do {
    defer { ka.finishEncoding() }
    try archival(ka)
}
```